File: hackerrank/codesprint/r1/a3.c
#include<stdio.h>

void cleartable(int broken[26][2])
{
	int i=0;
	for (i=0; i<26; i++)
	{
		broken[i][0]=0;
		broken[i][1]=0;
	}
}
int main() {
	int T;
	char c;
	int count;
	int broken[26][2];
	scanf("%d", &T);
	getchar();
	while (T>0) {
		cleartable(broken);
		count=0;
		while ((c=getchar())!='\n')
		{
			broken[c-97][0]=1;
		}
		while ((c=getchar())!='\n')
		{
			if (broken[c-97][0]==1)
			{
				if (broken[c-97][1]==0)
				{
					count++;
					broken[c-97][1]=1;
				}
			}
		}
		printf("%d\n", count);
		T=T-1;
	}

	return 0;
}

File: hackerrank/codesprint/r1/a3.py
def doaction():
	broken = raw_input()
	word = raw_input()
	count = 0
	done = []
	for each in word:
		if each in broken and each not in done:
			count = count + 1
			done.append(each)
	print count

def init():
	loops=raw_input()
	loops = int(loops)
	while loops>0:
		doaction()
		loops = loops - 1
	exit()
init()
